lib/acts_as_taggable_on_mongoid/tag_list.rb

```diff
@@ -12,6 +12,16 @@ module ActsAsTaggableOnMongoid
   #
   # If the input value(s) are to be parsed, then all values passed in are parsed.
   #
+  # Options:
+  #   parse   - True/False - indicates if all of the strings that are passed in are to be parsed
+  #             to split them into an array of strings.
+  #             Please note - if the passed in value is an array of strings, every string in the array
+  #             will be parsed.  If it is a single array, then just that string is parsed.
+  #   parser  - Class - A class that is used to parse the passed in strings.
+  #             If this parameter is supplied, parse is assumed to be truthy even if it is not passed in.
+  #   tagger  - object - An object that is to be used as the Tagger for the Taggable object.
+  #             This parameter is ignored if the tag does not support Taggers.
+  #
```

Comment:
I would prefer these be an explicit parse method rather than overload the constructor.
`TagList.new(tag_definition, "value 1", "value 2")`
`TagList.parse(tag_definition, "value 1, value 2")`
`TagList.parse(tag_definition, "value 1, value 2", "value 3, value 4", parser: ActsAsTaggableOnMongoid::GenericParser)`

Reply:
This pattern was inherited from ActsAsTaggable.
It supports/allows the assigning of values to a tag_list to be as flexible as possible by passing in parameters that indicate if the passed in value is to be parsed or not, and how.

We do not take advantage of or use that flexibility directly most of the time.

We do use it indirectly though, because allowing the parameters sets up the expectation and usage of the defaults for those parameters (setup through `ActsAsTaggableMongoid::configure` and/or the tag definition.)